### DISCUSSION TITLE
Force select query to go to Postgres master node in Kong

### DISCUSSION THREAD

**Discussion by anup-krai** [Kong/kong#8636] [Help] [answered]
We have Kong running in multiple datacenter (DC) with one master Postgres running in one DC and other replicas in other DCs. We are using oauth2 plugin for the API routes.

We see when token is generated a write call is made to master Postgres node and the same is getting synced to other replicas. So when actual API call comes then token will be read from read replicas and then API call is allowed.

In some cases we see before the tokens are synced to the replicas the API call is made and it’s getting rejected with HTTP 401.

How can we make the token validation call also to go to master Postgres instead of replicas?

Below is the code where the token is fetched -
https://github.com/Kong/kong/blob/master/kong/plugins/oauth2/access.lua#L790

**Comment by dndx** [comment, marked as answer]
Hello @anup-krai,

This is not something that can be done easily. Kong's config model is eventual consistent, and there is no way to guarantee that immediately after an Admin API call such change can take effect in the entire cluster (even if we force the validation to the same master as cache purge events are also propagated periodically only).

**Reply by anup-krai** [reply]
Thanks @dndx for your reply. Can you please share how Kong decides that a particular Postgres query is read or write and based on that it routes the calls to Postgres read or write node? 
Is there any parameter in the sql query which we can add which makes Kong to route the query to write Postgres node?

[Here](https://github.com/Kong/kong/blob/eaaae7c39d7964da458ddbedcac296a59de7e951/kong/db/strategies/postgres/connector.lua#L718) is one example  where operation is specified in sql query.

**Reply by anup-krai** [reply]
@dndx / Team, Is there any way to implement this?

**Reply by hanshuebner** [reply]
This is not something that we can support at this point.

**Reply by anup-krai** [reply]
sure, thanks @hanshuebner  for your response.
